mpi4py.spec

```diff
@@ -140,6 +140,7 @@ This package contains %{name} compiled against Open MPI.
 
 %if %{with_mpich}
 %package -n python%{python3_pkgversion}-mpi4py-mpich
+BuildRequires:  libfabric-devel
```

Comment:
Since this only applies to Leap 15, we should guard it with a check for that distro.

But moreover, this seems to me like it's an mpich packaging bug:

```
[2021-10-12T17:46:45.177Z] /usr/lib64/mpi/gcc/mpich/bin/mpicc -O2 -g -m64 -fmessage-length=0 -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables _configtest.o -Lbuild/temp.linux-x86_64-2.7 -o _configtest
[2021-10-12T17:46:45.177Z] /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: warning: libfabric.so.1, needed by /usr/lib64/libmercury_hl.so.2, not found (try using -rpath or -rpath-link)
```

Every package that wants to use mpich-devel shouldn't need to specify it's own `BuildRequires:  libfabric-devel`.  Probably the Leap 15 mpich-devel should have the `BuildRequires:  libfabric-devel`.  But it's additionally odd that this only seems to be needed on Leap 15.  Any thoughts on that?

Reply:
note that mpich does have that BR:
https://github.com/daos-stack/mpich/blob/daos_adio-rpm/mpich-LEAP_15.spec#L108

Reply:
> note that mpich does have that BR: https://github.com/daos-stack/mpich/blob/daos_adio-rpm/mpich-LEAP_15.spec#L108

It's _mpich-devel_ that needs a `Requires: libfabric-devel`.

Reply:
> for the change to guard this for a check for leap15, I'm not sure we want to do that because we don't guard it for other packages.

But it is only needed for leap15.  We shouldn't require packages to be installed that are not actually needed.

Reply:
agreed. what im saying is just that we should fix it for mpich and just remove this BR from all packages we have once that is fixed.